packages/nuxt/src/vite/sourceMaps.ts

```diff
@@ -1,52 +1,97 @@
-          filesToDeleteAfterUpload: sourceMapsUploadOptions.sourcemaps?.filesToDeleteAfterUpload ?? undefined,
-        },
-        _metaOptions: {
-          telemetry: {
-            metaFramework: 'nuxt',
-          },
-        },
-        debug: moduleOptions.debug ?? false,
-      });
+  nuxt.hook('vite:extendConfig', async (viteInlineConfig, _env) => {
+    if (sourceMapsEnabled && viteInlineConfig.mode !== 'development') {
+      const sentryPlugin = sentryVitePlugin(getPluginOptions(moduleOptions));
 
+      // Add Sentry plugin
       viteInlineConfig.plugins = viteInlineConfig.plugins || [];
       viteInlineConfig.plugins.push(sentryPlugin);
 
-      const sourceMapsPreviouslyEnabled = viteInlineConfig.build?.sourcemap;
+      // Enable source maps
+      viteInlineConfig.build = viteInlineConfig.build || {};
+      viteInlineConfig.build.sourcemap = true;
 
-      if (moduleOptions.debug && !sourceMapsPreviouslyEnabled) {
-        // eslint-disable-next-line no-console
-        console.log('[Sentry]: Enabled source maps generation in the Vite build options.');
-        if (!moduleOptions.sourceMapsUploadOptions?.sourcemaps?.filesToDeleteAfterUpload) {
-          // eslint-disable-next-line no-console
-          console.warn(
-            `[Sentry] We recommend setting the \`sourceMapsUploadOptions.sourcemaps.filesToDeleteAfterUpload\` option to clean up source maps after uploading.
-[Sentry] Otherwise, source maps might be deployed to production, depending on your configuration`,
-          );
+      logDebugInfo(moduleOptions, viteInlineConfig.build?.sourcemap);
+    }
+  });
+
+  nuxt.hook('nitro:config', (nitroConfig: NitroConfig) => {
+    if (sourceMapsEnabled && !nitroConfig.dev) {
+      const sentryPlugin = sentryRollupPlugin(getPluginOptions(moduleOptions));
+
+      if (nitroConfig.rollupConfig) {
+        // Add Sentry plugin
+        if (!Array.isArray(nitroConfig.rollupConfig.plugins)) {
+          nitroConfig.rollupConfig.plugins = nitroConfig.rollupConfig.plugins ? [nitroConfig.rollupConfig.plugins] : [];
         }
-      }
+        nitroConfig.rollupConfig.plugins.push(sentryPlugin);
 
-      viteInlineConfig.build = viteInlineConfig.build || {};
-      viteInlineConfig.build.sourcemap = true;
+        // Enable source maps
+        nitroConfig.rollupConfig.output = nitroConfig?.rollupConfig?.output || {};
+        nitroConfig.rollupConfig.output.sourcemap = true;
+        nitroConfig.rollupConfig.output.sourcemapExcludeSources = false; // Adding "sourcesContent" to the source map (Nitro sets this eto `true`)
+
+        logDebugInfo(moduleOptions, nitroConfig.rollupConfig.output?.sourcemap);
+      }
     }
   });
 }
+
+/**
+ * Normalizes the beginning of a path from e.g. ../../../ to ./
+ */
+function normalizePath(path: string): string {
+  return path.replace(/^(\.\.\/)+/, './');
+}
+
+function getPluginOptions(moduleOptions: SentryNuxtModuleOptions): object {
```

Comment:
h: Can we type the return value properly? Typing things as `object` is always sus.